soda/duckdb/soda/data_sources/duckdb_data_source.py

```diff
@@ -0,0 +1,130 @@
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+
+from soda.common.exceptions import DataSourceConnectionError
+from soda.common.logs import Logs
+from soda.execution.data_source import DataSource
+from soda.execution.data_type import DataType
+
+logger = logging.getLogger(__name__)
+
+
+class DuckDBCursor:
+    def __init__(self, connection):
+        self._connection = connection
+
+    def __getattr__(self, attr):
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        return getattr(self._connection, attr)
+
+    def close(self):
+        # because a duckdb cursor is actually the current connection,
+        # we don't want to close it
+        pass
+
+
+class DuckDBDataSourceConnectionWrapper:
+    def __init__(self, delegate):
+        self._delegate = delegate
+
+    def __getattr__(self, attr):
+        if attr in self.__dict__:
+            return getattr(self, attr)
+        return getattr(self._delegate, attr)
+
+    def cursor(self):
+        return DuckDBCursor(self._delegate)
+
+
+class DuckDBDataSource(DataSource):
+    TYPE = "duckdb"
+
+    # Maps synonym types for the convenience of use in checks.
+    # Keys represent the data_source type, values are lists of "aliases" that can be used in SodaCL as synonyms.
+    SCHEMA_CHECK_TYPES_MAPPING: dict = {
+        "character varying": ["varchar"],
+        "double precision": ["double"],
+        "timestamp without time zone": ["timestamp"],
+        "decimal": ["decimal(18,3)"],
+    }
+
+    SQL_TYPE_FOR_CREATE_TABLE_MAP: dict = {
+        DataType.TEXT: "varchar",
+        DataType.INTEGER: "integer",
+        DataType.DECIMAL: "decimal",
+        DataType.DATE: "date",
+        DataType.TIME: "time",
+        DataType.TIMESTAMP: "timestamp",
+        DataType.TIMESTAMP_TZ: "timestamp with time zone",
+        DataType.BOOLEAN: "boolean",
+    }
+
+    SQL_TYPE_FOR_SCHEMA_CHECK_MAP = {
+        DataType.TEXT: "varchar",
+        DataType.INTEGER: "integer",
+        DataType.DECIMAL: "decimal(18,3)",
+        DataType.DATE: "date",
+        DataType.TIME: "time",
+        DataType.TIMESTAMP: "timestamp",
+        DataType.TIMESTAMP_TZ: "timestamp with time zone",
+        DataType.BOOLEAN: "boolean",
+    }
+
+    NUMERIC_TYPES_FOR_PROFILING = [
+        "tinyint",
+        "smallint",
+        "integer",
+        "bigint",
+        "decimal",
+        "decimal(18,3)",
+        "double",
+        "real",
+    ]
+    TEXT_TYPES_FOR_PROFILING = ["char", "varchar"]
+
+    def __init__(self, logs: Logs, data_source_name: str, data_source_properties: dict):
+        super().__init__(logs, data_source_name, data_source_properties)
+        self.path = data_source_properties.get("path")
+        self.read_only = data_source_properties.get("read_only", False)
+        self.duckdb_connection = data_source_properties.get("duckdb_connection")
```

Comment:
Not a blocker, but could you add a parameter for [configuration](https://duckdb.org/docs/sql/configuration.html)?
Users might want to have specific configuration when running soda checks.

Reply:
@JorgeGarciaIrazabal  Can you please open an issue for it - we can tackle as a separate PR